src/query-builder/SelectQueryBuilder.ts

```diff
@@ -3410,7 +3410,8 @@ export class SelectQueryBuilder<Entity extends ObjectLiteral>
         // first query find ids in skip and take range
         // and second query loads the actual data in given ids range
         if (
-            (this.expressionMap.skip || this.expressionMap.take) &&
+            this.expressionMap.skip &&
+            this.expressionMap.take &&
```

Comment:
This would make the occurence of both skip and take necessary. instead of either of them. But there are cases where only one of them might be there and then pagination would fail

Reply:
All of the `findOne` versions add a `{ take: 1 }`, so "either of them" is incorrect as well. If I'm using an offset without a limit, then I'm not really taking a page. It's possible that no condition based on skip and take is correct as well. I just want to make people start looking at it.